### PR TITLE
Normalize json.RawMessage in writeJSON and writeQuiet

### PR DESCRIPTION
**Pull request by jeremy**
## Summary

- `writeJSON` and `writeQuiet` passed data directly to `json.Encoder.Encode` without normalizing `json.RawMessage` first, while all other output paths (`writeStyled`, `writeIDs`, `writeCount`, `writeLiteralMarkdown`) already called `NormalizeData` before rendering
- Adds `NormalizeData` calls in `writeJSON` (for `--json` mode) and `writeQuiet` (for `--agent`/`--quiet` mode) to fix `MarshalJSON` failures on encoding edge cases

## Test plan

- [x] `go test ./internal/output/...` passes
- [x] `TestWriteJSONWithRawMessage` — verifies FormatJSON normalizes `json.RawMessage` in `Response.Data`
- [x] `TestWriteQuietWithRawMessage` — verifies FormatQuiet normalizes bare `json.RawMessage`

### CODE REVIEW

File: internal/output/envelope.go [hunk 1]
```diff
@@ -191,6 +191,9 @@ func isTTY(w io.Writer) bool {
 }
 
 func (w *Writer) writeJSON(v any) error {
+	if resp, ok := v.(*Response); ok {
+		resp.Data = NormalizeData(resp.Data)
+	}
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/output/envelope.go:201]: Good catch on the precision concern. This is a pre-existing characteristic of `NormalizeData` — all other output paths (`writeStyled`, `writeIDs`, `writeCount`, `writeLiteralMarkdown`) already go through the same `json.Unmarshal` without `UseNumber`. Currently `json.RawMessage` only flows from the quickstart command with small numbers, not from API responses, so there's no practical impact today. Adding `UseNumber` to `NormalizeData` would be a good hardening improvement but it's a broader change across all output paths — I'll track that separately.

File: internal/output/envelope.go [hunk 2]
```diff
@@ -191,6 +191,9 @@ func isTTY(w io.Writer) bool {
...
 	enc := json.NewEncoder(w.opts.Writer)
 	enc.SetIndent("", "  ")
 	return enc.Encode(v)
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/output/envelope.go]: Fixed — switched to shallow copy pattern to match the convention used by all other call sites (`writeIDs`, `writeCount`, `presentStyledEntity`, `presentMarkdownEntity`, both renderers).

File: internal/output/envelope.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/output/envelope.go:201]: Fixed — `NormalizeData` now uses `json.NewDecoder` with `UseNumber()` so numbers stay as `json.Number` instead of being converted to `float64`. Added `json.Number` handling to `formatCell` and `isZeroValue` downstream, and tests verifying large integers (>53-bit) survive the round trip.
